### PR TITLE
Adding califorest model

### PR DESCRIPTION
**Pull request by Pallavim9**
### Califorest Model for PyHealth Contribution
Pallavi Maddineni (pm35) & Gokul Naidu (gnaidu2)

### Original Paper
Y. Park and J. C. Ho. 2020. CaliForest: Calibrated Random Forest for Health Data. ACM Conference on Health, Inference, and Learning (2020)
https://pubmed.ncbi.nlm.nih.gov/34308443/
https://github.com/yubin-park/califorest

### Description
This PR adds a Califorest class, which wraps a scikit-learn RandomForestClassifier with calibration using CalibratedClassifierCV. 

##### Example Usage
```
# Initialize the model
model = Califorest(
    n_estimators=300,
    criterion='gini',
    max_depth=5,
    min_samples_split=2,
    min_samples_leaf=1,
    ctype='isotonic'
)

# Fit the model
model.fit(X_train, y_train)

# Make predictions
y_pred = model.predict(X_test)
```

### CODE REVIEW

**Comment by jhnwu3** [comment]
Closing this PR as it lacks proper labeling. Please add appropriate labels and reopen if needed.
